src/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -0,0 +1,1755 @@
+                return _portList;
+            }
+        }
+
+        private string _Port
+        {
+            get
+            {
+                return _portImplicit ? string.Empty :
+                      (SpecialAttributeLiteral
+                       + PortAttributeName
+                       + ((_port.Length == 0) ? string.Empty : (EqualsLiteral + _port)));
+            }
+        }
+
+        public bool Secure
+        {
+            get
+            {
+                return _secure;
+            }
+            set
+            {
+                _secure = value;
+            }
+        }
+
+        public DateTime TimeStamp
+        {
+            get
+            {
+                return _timeStamp;
+            }
+        }
+
+        public string Value
+        {
+            get
+            {
+                return _value;
+            }
+            set
+            {
+                _value = value ?? string.Empty;
+            }
+        }
+
+        internal CookieVariant Variant
+        {
+            get
+            {
+                return _cookieVariant;
+            }
+            set
+            {
+                // Only set by HttpListenerRequest::Cookies_get()
+#if !NETNative_SystemNetHttp
+                GlobalLog.Assert(value == CookieVariant.Rfc2965, "Cookie#{0}::set_Variant()|value:{1}", Logging.HashString(this), value);
+#endif
+                _cookieVariant = value;
+            }
+        }
+
+        // _domainKey member is set internally in VerifySetDefaults().
+        // If it is not set then verification function was not called;
+        // this should never happen.
+        internal string DomainKey
+        {
+            get
+            {
+                return _domainImplicit ? Domain : _domainKey;
+            }
+        }
+
+        public int Version
+        {
+            get
+            {
+                return _version;
+            }
+            set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException("value");
+                }
+                _version = value;
+                if (value > 0 && _cookieVariant < CookieVariant.Rfc2109)
+                {
+                    _cookieVariant = CookieVariant.Rfc2109;
+                }
+            }
+        }
+
+        private string _Version
+        {
+            get
+            {
+                return (Version == 0) ? string.Empty :
+                                       (SpecialAttributeLiteral
+                                       + VersionAttributeName
+                                       + EqualsLiteral + (IsQuotedVersion ? "\"" : string.Empty)
+                                       + _version.ToString(NumberFormatInfo.InvariantInfo) + (IsQuotedVersion ? "\"" : string.Empty));
+            }
+        }
+
+        internal static IComparer GetComparer()
```

Comment:
I've removed this method in favor of a CookieComparer singleton provided by a property on that type.